packages/xdl/src/project/createBundlesAsync.ts

```diff
@@ -24,17 +24,59 @@ type PackagerOptions = {
   minify: boolean;
 };
 
-export function printBundleSizes(bundles: { android: BundleOutput; ios: BundleOutput }) {
-  const files = [
-    ['index.ios.js', bundles.ios.code],
-    ['index.android.js', bundles.android.code],
-  ];
-  // Account for inline source maps
-  if (bundles.ios.map) {
-    files.push([chalk.dim('index.ios.js.map'), bundles.ios.map]);
-  }
-  if (bundles.android.map) {
-    files.push([chalk.dim('index.android.js.map'), bundles.android.map]);
+// If `isHermesPreferable` is true, will try use hermes bytecode bundle as primary output.
+// This if for classic `expo export` or `expo publish` flows.
+export function printBundleSizes(
+  bundles: { android: BundleOutput; ios: BundleOutput },
+  isHermesPreferable: boolean
+) {
+  const files: [string, string | Uint8Array][] = [];
+
+  if (isHermesPreferable) {
+    if (bundles.ios.hermesBytecodeBundle) {
```

Comment:
```suggestion
  if (isHermesPreferable && bundles.ios.hermesBytecodeBundle) {
```
If we do this for each of the if/else statements, can we get rid of half of these?

Reply:
if the else block is needed, this style may not benefit a lot ?
let me answer why we need the else block below.

Reply:
Sounds good. Agreed that this comment only applies if we can get rid of the else block.

Reply:
let me remove `isHermesPreferable` entirely as we don't need both hermes bytecode and plaintext js bundle.